ABAC.DAL/Services/ResourceService.cs

```diff
@@ -93,8 +93,18 @@ public async Task DeleteAttributeAsync(int id, string attributeName)
                 throw new NotFoundException();
             }
 
-            resource.Attributes.Remove(attributeName);
-            await repository.CreateOrUpdateAsync(resource);
+			var attribute = from attr in resource.Attributes
```

Comment:
I suggest to change it to `resource.Attributes.SingleOrDefault(a => a.Name == attributeName)` to maintain consistency.